File: 은지/LV1/소수 찾기.js
function solution(n) {
  // 배수 확인 배열
  var check = []
  for (let i=0; i<=n; i++) {
    check.push(0)
  }
  // 소수 배열
  var prime = []
  for (let x=2; x<=n; x++) {
    // x가 소수
    if (check[x] === 0) {
      // 소수 리스트 추가
      prime.push(x)
      // x의 배수 체크
      for (let y=x; y<=n; y+=x) {
        check[y] = 1
      }
    }
  }
  // 소수 개수
  const answer = prime.length
  return answer;
}

File: 은지/LV2/소수 찾기.py
import itertools

def prime(x):
    if x < 2:
        return 0
    else:
        for i in range(2, x):
            if x % i == 0:
                return 0
        return 1

def solution(numbers):
    answer = 0
    a = []
    lst = [str(x) for x in numbers]
    lst.sort(reverse=True)

    arr = [x for x in range(len(lst))]
    for x in itertools.permutations(arr, len(lst)):
        sub_s = ''
        for i in range(len(lst)):
            sub_s += lst[x[i]]
            if prime(int(sub_s)) and int(sub_s) not in a:
                a.append(int(sub_s))
    answer = len(a)
    return answer

numbers = "011"
print(solution(numbers))